apps/prairielearn/src/lib/workspace.js

```diff
@@ -451,6 +496,39 @@ module.exports = {
       }
     }
 
+    if (fileGenerationErrors.length > 0) {
+      const output = fileGenerationErrors.map((error) => `${error.file}: ${error.msg}`).join('\n');
+      issues.insertIssue({
+        variantId: variant.id,
+        studentMessage: 'Error retrieving workspace files',
+        instructorMessage: 'Error retrieving workspace files',
+        manuallyReported: false,
+        courseCaused: true,
+        courseData: { workspace, variant, question, course },
+        systemData: {
+          courseErrData: {
+            // This is shown in the console log of the issue
+            outputBoth: output,
+            // This data is only shown if user is admin (e.g., in dev mode).
+            errors: fileGenerationErrors.map((error) => ({
+              ...error,
+              // Since error is typically not serializable, a custom object is created.
+              err: error.err
+                ? {
+                    ...error.err,
+                    stack: error.err.stack,
+                    data: error.err.data,
+                    message: error.err.message,
+                    cause: error.err.cause,
+                  }
+                : undefined,
+            })),
+          },
+        },
+        authnUserId: null,
```

Comment:
This is in discussion on Slack, just tagging it here as pending confirmation that it will indeed remain null.